### PR TITLE
fix(ingest/unity): simplify MCP generation and reporting

### PR DESCRIPTION
**Pull request by hsheth2**
## Checklist

- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 1]
```diff
@@ -294,22 +301,15 @@ def _generate_column_lineage_mcp(
             upstreams.append(upstream_table)
 
         if upstreams:
-            upstream_lineage = UpstreamLineageClass(
+            return UpstreamLineageClass(
```
Comment (jjoyce0510) [inline metadata-ingestion/src/datahub/ingestion/source/unity/source.py:304]: nice simplification

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py [hunk 2]
```diff
@@ -224,37 +225,43 @@ def process_tables(self, schema: proxy.Schema) -> Iterable[MetadataWorkUnit]:
...
             if self.config.include_column_lineage:
                 self.unity_catalog_api_proxy.get_column_lineage(table)
-                yield from self._generate_column_lineage_mcp(dataset_urn, table)
+                lineage = self._generate_column_lineage_aspect(dataset_urn, table)
             else:
                 self.unity_catalog_api_proxy.table_lineage(table)
-                yield from self._generate_lineage_mcp(dataset_urn, table)
+                lineage = self._generate_lineage_aspect(dataset_urn, table)
+
+            yield from [
```
[the remark this answers is not in the document]
Comment (hsheth2) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:251]: yes - it felt more clear to have all the processing first, and then just one main `yield from`

it also avoids repeating the as_workunit() stuff all over the place

**Review by jjoyce0510 — APPROVED** [review]
LGTM.

File: metadata-ingestion/src/datahub/ingestion/source/unity/source.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (jjoyce0510) [reply metadata-ingestion/src/datahub/ingestion/source/unity/source.py:251]: makes sense

**Pull request merged**